frontend/src/components/file/FileSelectModal.vue

```diff
@@ -0,0 +1,89 @@
+<script setup lang="ts">
+import { computed } from 'vue';
+import DocumentCard from '@/components/file/DocumentCard.vue';
+import { Button, Dialog } from '@/lib/primevue';
+
+import type { Ref } from 'vue';
+import type { Document } from '@/types';
+
+// Props
+type Props = {
+  documents: Array<Document>;
+  selectedDocuments?: Array<Document>;
+};
+
+const props = withDefaults(defineProps<Props>(), {
+  selectedDocuments: () => []
+});
+
+// Emits
+const emit = defineEmits(['fileSelect:submit']);
+
+// State
+const visible = defineModel<boolean>('visible');
+const selectedFiles: Ref<Array<Document>> = computed(() => props.selectedDocuments);
+
+// Actions
+function onDocumentClicked(data: { document: Document; selected: boolean }) {
+  if (data.selected) {
+    if (!selectedFiles.value.includes(data.document)) {
```

Comment:
Might be able to collapse these conditionals together into one if statement if the converse else statements are fine being executed.